task/github-create-deployment-status/0.1/github-create-deployment-status.yaml

```diff
@@ -0,0 +1,152 @@
+      description: |
+        Adds a new inactive status to all prior non-transient,
+        non-production environment deployments with the same repository and
+        environment name as the created status's deployment.
+      default: "true"
+      type: string
+
+    - name: GITHUB_TOKEN_SECRET_NAME
+      description: |
+        The name of the Kubernetes Secret that contains the GitHub token.
+      default: "github"
+      type: string
+
+    - name: GITHUB_TOKEN_SECRET_KEY
+      description: |
+        The key within the Kubernetes Secret that contains the GitHub token.
+      default: "token"
+      type: string
+
+  steps:
+    - name: create-deployment-status
+      env:
+        - name: GITHUBTOKEN
+          valueFrom:
+            secretKeyRef:
+              name: $(params.GITHUB_TOKEN_SECRET_NAME)
+              key: $(params.GITHUB_TOKEN_SECRET_KEY)
+
+      image: registry.access.redhat.com/ubi8/ubi:8.2
+      script: |
+        #!/usr/libexec/platform-python
+        import http.client
+        import json
+        import os
+        import sys
+
+        deployment_status_url = \
+            "$(params.API_PATH_PREFIX)" + \
+            "/repos/$(params.REPO_FULL_NAME)/deployments" + \
+            "/$(params.DEPLOYMENT_ID)/statuses"
+
+        data = {
+            "state": "$(params.STATE)",
+            "log_url": "$(params.LOG_URL)",
+            "description": "$(params.DESCRIPTION)",
+            "environment": "$(params.ENVIRONMENT)",
+            "environment_url": "$(params.ENVIRONMENT_URL)",
+            "auto_inactive": json.loads("$(params.AUTO_INACTIVE)"),
+        }
+
+        print("Sending this data to GitHub: ")
+        print(data)
+
+        # This is for our fake github server
+        if "$(params.GITHUB_HOST_URL)".startswith("http://"):
+          conn = http.client.HTTPConnection("$(params.GITHUB_HOST_URL)".replace("http://", ""))
+        else:
+          conn = http.client.HTTPSConnection("$(params.GITHUB_HOST_URL)")
+
+        r = conn.request(
+            "POST",
+            deployment_status_url,
+            body=json.dumps(data),
+            headers={
+                "User-Agent": "TektonCD, the peaceful cat",
+                "Authorization": "Bearer " + os.environ["GITHUBTOKEN"],
+                "Accept": "application/vnd.github.ant-man-preview+json, " +
+                          "application/vnd.github.flash-preview+json"
+            })
+        resp = conn.getresponse()
+
+        if resp.status != 201:
+            print("Error: %d" % (resp.status))
+            print(resp.read())
+            sys.exit(1)
+        else:
+            body = json.loads(resp.read().decode())
+            print("GitHub deployment status created for "
+                  "$(params.REPO_FULL_NAME)/$(params.DEPLOYMENT_ID): "
+                  "id={} environment={} state={}".format(
+                        body["id"],
+                        body["environment"],
+                        body["state"]))
```

Comment:
apart from printing in the logs do we need to output something as results so that other tasks can consume the result ?

Reply:
I think we don't need to set result since their is no mutating operation API on deployment status.

BTW, I think setting `id` or `url` as a result for deployment (related: #675) can be used as parameter for this task, WDYT?

Reply:
SGTM

Reply:
I added `ID`, `URL` and `STATUSES_URL` as a result to `github-create-deployment` 🙂

https://github.com/tektoncd/catalog/pull/675/files#diff-045f5d87b51ea5167320c12f3b35aafdbb2c8e11cfd49584ba1465e7b25ff619R89-R96